File: simple_fix.py
#!/usr/bin/env python3
"""
Simple script to fix the most common syntax errors
"""
import os
import re
import glob

def fix_syntax_errors(file_path):
    """Fix syntax errors in a file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        
        # Fix malformed class declarations
        content = re.sub(r'class\s+(\w+)\s+extends\s+Component<[^>]+>\s*/>', r'class \1 extends Component<\2>', content)
        
        # Fix malformed state declarations
        content = re.sub(r'public\s+state:\s*const\s+(\w+)\s*=\s*\{,\s*', r'public state: \1 = {\n    ', content)
        
        # Fix malformed JSX attributes with double braces
        content = re.sub(r'style=\{\{\s*([^}]+)\s*\}\}', r'style={{\1}}', content)
        
        # Fix malformed className attributes - add spaces between concatenated classes
        content = re.sub(r'className="([^"]*?)([a-zA-Z])([A-Z])', r'className="\1\2 \3', content)
        
        # Fix malformed JSX closing tags
        content = re.sub(r'<(\w+)\s*/>', r'</\1>', content)
        
        # Fix missing spaces in style values
        content = re.sub(r'(\w+):\s*\'([^\']*?)([a-zA-Z])([a-zA-Z])', r'\1: \'\2\3 \4', content)
        
        # Clean up extra whitespace
        content = re.sub(r'\n\s*\n\s*\n', '\n\n', content)
        
        # Write back if changed
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            return True
        
        return False
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to process all files"""
    # Focus on the most problematic files first
    target_files = [
        'app/components/EnhancedSEOHead.tsx',
        'app/components/ErrorBoundary.tsx',
        'app/components/LoadingStates.tsx'
    ]
    
    files_processed = 0
    files_fixed = 0
    
    for file_path in target_files:
        if os.path.exists(file_path):
            files_processed += 1
            if fix_syntax_errors(file_path):
                files_fixed += 1
                print(f"Fixed syntax errors in: {file_path}")
    
    print(f"\nProcessed {files_processed} files")
    print(f"Fixed syntax errors in {files_fixed} files")

if __name__ == "__main__":
    main()

File: simple_fix.js

#!/usr/bin/env node
const fs = require('fs')
const path = require('path')
// Function to fix merge conflicts
function fixMergeConflicts(content) {
  // Remove merge conflict markers and choose the HEAD version
  return content
    .replace(/<<<<<<< HEAD\n([\s\S]*?)=======\n([\s\S]*?)>>>>>>> [^\n]+\n/g, '$1')
    .replace(/<<<<<<< HEAD:([^\n]+)\n([\s\S]*?)=======\n([\s\S]*?)>>>>>>> [^\n]+\n/g, '$2')
    .replace(/<<<<<<< HEAD\n([\s\S]*?)=======\n([\s\S]*?)>>>>>>> [^\n]+/g, '$1')
}
// Function to fix common syntax errors
function fixSyntaxErrors(content) {
  // Fix missing closing tags for fragments
  content = content.replace(/<>\s*$/gm, '<>')
  content = content.replace(/^\s*<\/>/gm, '</>')
  // Fix missing commas in object literals
  content = content.replace(/(\w+):\s*([^}\n]+)\s*\n/g, '$1: $2,\n')
  // Fix missing semicolons
  content = content.replace(/([^;}])\s*$/gm, '$1;')
  // Fix JSX expressions that need wrapping
  content = content.replace(/return\s*\(\s*([^<]+)\s*\)/g, (match, content) => {
    if (content.trim() && !content.includes('<')) {
      return `return (\n    <>\n      ${content}\n    </>\n  )`
    }
    return match
  })
  return content
}
// Function to fix specific component issues
function fixComponentIssues(content) {
  // Remove duplicate imports
  const lines = content.split('\n')
  const importLines = []
  const otherLines = []
  for (const line of lines) {
    if (line.trim().startsWith('import ') && line.includes(' from ')) {
      if (!importLines.includes(line)) {
        importLines.push(line)
      }
    } else {
      otherLines.push(line)
    }
  }
  // Reconstruct content
  content = importLines.join('\n') + '\n' + otherLines.join('\n')
  // Fix console statements
  content = content.replace(/console\.(log|warn|error)/g, '// eslint-disable-next-line no-console\n    console.$1')
  return content
}
// Function to process a file
function processFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8')
    const originalContent = content
    // Apply fixes
    content = fixMergeConflicts(content)
    content = fixSyntaxErrors(content)
    content = fixComponentIssues(content)
    // Only write if content changed
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8')
      // eslint-disable-next-line no-console
    console.log(`Fixed: ${filePath}`)
      return true
    }
    return false
  } catch (error) {
    // eslint-disable-next-line no-console
    console.error(`Error processing ${filePath}:`, error.message)
    return false
  }
}
// Function to find all relevant files
function findFiles(dir, extensions = ['.tsx', '.ts', '.js']) {
  const files = []
  function traverse(currentDir) {
    try {
      const items = fs.readdirSync(currentDir)
      for (const item of items) {
        const fullPath = path.join(currentDir, item)
        const stat = fs.statSync(fullPath)
        if (stat.isDirectory()) {
          // Skip node_modules and .next
          if (!['node_modules', '.next', '.git'].includes(item)) {
            traverse(fullPath)
          }
        } else if (stat.isFile()) {
          const ext = path.extname(item)
          if (extensions.includes(ext)) {
            files.push(fullPath)
          }
        }
      }
    } catch (error) {
  // Skip directories we can't read
}
  }
  traverse(dir)
  return files
}
// Main execution
function main() {
  const workspaceDir = process.cwd()
  // eslint-disable-next-line no-console
    console.log('Starting error fixing...')
  // Find all relevant files
  const files = findFiles(workspaceDir)
  // eslint-disable-next-line no-console
    console.log(`Found ${files.length} files to process`)
  let fixedCount = 0
  // Process each file
  for (const file of files) {
  if (processFile(file)) {
      fixedCount++
}
  }
  // eslint-disable-next-line no-console
    console.log(`\nFixed ${fixedCount} files`)
  // eslint-disable-next-line no-console
    console.log('Error fixing completed!')
}
// Run the script
if (require.main === module) {
  main()
}
module.exports = { fixMergeConflicts, fixSyntaxErrors, fixComponentIssues, processFile }